packages/data-stores/src/site/selectors.ts

```diff
@@ -113,3 +117,22 @@ export const requiresUpgrade = ( state: State, siteId: number | null ) => {
 
 	return Boolean( ! isWoopFeatureActive && hasWoopFeatureAvailable );
 };
+
+export function isJetpackSite( state: State, siteId?: number ): boolean {
+	return Boolean( siteId && select( STORE_KEY ).getSite( siteId )?.jetpack );
+}
+
+export function isEligibleForProPlan( state: State, siteId?: number ): boolean {
+	if ( ! siteId ) {
+		return false;
+	}
+
+	if (
+		( isJetpackSite( state, siteId ) && ! isSiteAtomic( state, siteId ) ) ||
+		isSiteWPForTeams( state, siteId )
+	) {
+		return false;
+	}
```

Comment:
Stolen from https://github.com/Automattic/wp-calypso/blob/update/stepper-pro-plan/client/my-sites/plans-comparison/is-eligible-for-pro-plan.ts#L10-L12